app/models/stat/anime_amount_watched.rb

```diff
@@ -0,0 +1,72 @@
+class Stat
+  class AnimeAmountWatched < Stat
+    # fully regenerate data.
+    def recalculate!
+      entries = user.library_entries.eager_load(:anime)
+                    .where.not('anime.episode_length': nil)
+
+      stats_data['all_time'] = {
+        total_anime: entries.count, # all anime
+        total_episodes: entries.sum(:progress), # all episodes
+        total_time: entries.sum(:time_spent) # time spent
+      }
+
+      save!
+    end
+
+    def self.increment(user, library_entry)
+      record = user.stats.find_or_initialize_by(
+        type: 'Stat::AnimeAmountWatched'
+      )
+      record = record.default_stats if record.new_record?
+      record.increment(library_entry)
+    end
+
+    def increment(library_entry)
+      stats_data['all_time']['total_anime'] += 1
+      stats_data['all_time']['total_episodes'] +=
+        progress_difference(library_entry)
+      stats_data['all_time']['total_time'] +=
+        progress_to_time(library_entry, progress_difference(library_entry))
+
+      save!
+    end
+
+    def self.decrement(user, library_entry)
+      record = user.stats.find_by(type: 'Stat::AnimeAmountWatched')
+      return unless record
+      record.decrement(library_entry)
+    end
+
+    def decrement(library_entry)
+      stats_data['all_time']['total_anime'] -= 1
+      stats_data['all_time']['total_episodes'] -= library_entry.progress
+      stats_data['all_time']['total_time'] -=
+        progress_to_time(library_entry, library_entry.progress)
+
+      save!
+    end
+
+    def default_stats
+      # needs to be a string and NOT a symbol
+      # due to how it is being saved.
```

Comment:
I'm pretty sure Rails will handle all the conversions itself?

Also it's probably better for `default_stats` to just be a `DEFAULT_STATS` constant that you `self.stats_data = DEFAULT_STATS if stats_data.blank?`, instead of a method which modifies the instance and returns the self.

When a method has a name like `default_stats`, you expect that it'll return the default stats.  The fact that this method doesn't is a bit of a betrayal of expectations.